visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts

```diff
@@ -26,23 +26,35 @@ export class FileChooserController {
 	public onImportNewFiles(element) {
 		this.$scope.$apply(() => {
 			this.storeService.dispatch(resetFiles())
+			let content
+
 			for (const file of element.files) {
+				const isCompressed = file.name.endsWith(".gz")
 				const reader = new FileReader()
 				reader.onloadstart = () => {
 					this.storeService.dispatch(setIsLoadingFile(true))
 				}
 				reader.onload = event => {
-					this.setNewData(file.name, (<any>event.target).result)
+					if (isCompressed) {
+						const zlib = require("zlib")
+
+						const buffer = new Buffer((<any>event.target).result, "UTF-8")
```

Comment:
This should cause a deprecation warning while using it? Using `new Buffer()` could result in security vulnerabilities and should therefore definitely not be used. Instead, please use `Buffer.from()` as described 
here: https://nodejs.org/api/buffer.html

Reply:
Oh you're right! It's deprecated. I changed it :)